cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -1500,6 +1500,9 @@ public void deleteState(AppStateKey request) throws ApplicationNotFoundException
    */
   public void deleteAllStates(NamespaceId namespaceId, String appName)
       throws ApplicationNotFoundException {
+
+    accessEnforcer.enforce(new ApplicationId(namespaceId.getNamespace(), appName),
+        authenticationContext.getPrincipal(), StandardPermission.UPDATE);
```

Comment:
I'm assuming here that deleting application state does not delete the application itself. If that's the case, LGTM.

Reply:
Yes, deleting state does not delete the app. This will be useful when a user chooses not to use saved data especially if the saved data is outdated.